src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,497 @@
+      else if (expr instanceof GoAddExpr) {
+        result.append(getValue(expr));
+      }
+    }
+
+    return result.toString();
+  }
+
+  private static int getPlaceholderPosition(@NotNull GoFunctionOrMethodDeclaration function) {
+    Integer position = FORMATTING_FUNCTIONS.get(StringUtil.toLowerCase(function.getName()));
+    if (position != null) {
+      String importPath = function.getContainingFile().getImportPath(false);
+      if ("fmt".equals(importPath) || "log".equals(importPath) || TESTING_PATH.equals(importPath)) {
+        return position;
+      }
+    }
+    return -1;
+  }
+
+  @Nullable
+  private static String resolve(@NotNull GoExpression argument) {
+    if (argument instanceof GoStringLiteral) return argument.getText();
+
+    PsiReference reference = argument.getReference();
+    PsiElement resolved = reference != null ? reference.resolve() : null;
+
+    if (resolved instanceof GoVarDefinition) {
+      return getValue(((GoVarDefinition)resolved).getValue());
+    }
+    if (resolved instanceof GoConstDefinition) {
+      return getValue(((GoConstDefinition)resolved).getValue());
+    }
+
+    return null;
+  }
+
+  public void checkPrint() {
+    // TODO florin: Implement the print function checking
+    // Ref code: https://github.com/golang/go/blob/79f7ccf2c3931745aeb97c5c985b6ac7b44befb4/src/cmd/vet/print.go#L587
+  }
+
+  private class FormatState {
+    public char verb;             // the format verb: 'd' for "%d"
+    public String format;         // the full format directive from % through verb, "%.3d".
+    public String flags;          // the list of # + etc.
+    public List<Integer> argNums; // the successive argument numbers that are consumed, adjusted to refer to actual arg in call
+    public boolean indexed;       // whether an indexing expression appears: %[1]d.
+    public int firstArg;          // Index of first argument after the format in the Printf call.
+
+    // Used only during parse.
+    public int argNum;            // Which argument we're expecting to format now.
+    public boolean indexPending;  // Whether we have an indexed argument that has not resolved.
+    public int nBytes;            // number of bytes of the format string consumed.
+
+    GoExpression myPlaceholder;
+
+    public FormatState(GoExpression placeholder) {
+      myPlaceholder = placeholder;
+    }
+
+    public void parseFlags() {
+      String knownFlags = "#0+- ";
+      StringBuilder flags = new StringBuilder(this.flags);
+      for (; nBytes < format.length(); nBytes++) {
+        if (knownFlags.indexOf(format.charAt(nBytes)) != -1) {
+          flags.append(format.charAt(nBytes));
+        }
+        else {
+          this.flags = flags.toString();
+          return;
+        }
+      }
+      this.flags = flags.toString();
+    }
+
+    private void scanNum() {
+      for (; nBytes < format.length(); nBytes++) {
+        if (format.charAt(nBytes) < '0' || '9' < format.charAt(nBytes)) {
+          return;
```

Comment:
From what I understand in the method documentation, Character#isDigit will match also non-Latin digits. Go is explicit at supporting only ` 0-9 ` digits for everything here so we could even add a "Number is not a valid Latin-1 digit" error here. What do you think?